sources/api/apiclient/src/network.rs

```diff
@@ -0,0 +1,221 @@
+//! This module allows configuration of network settings from URIs or stdin. The input is expected to be
+//! a TOML network configuration file in netdog format. The configuration is written directly to
+//! /.bottlerocket/net.toml for validation and application at next boot.
+
+use base64::{engine, Engine};
+use reqwest::Url;
+use snafu::{futures::TryFutureExt, OptionExt, ResultExt};
+use std::path::Path;
+use tokio::io::AsyncReadExt;
+
+/// Configures network settings from the given input source (URI or stdin), writing the content
+/// directly to /.bottlerocket/net.toml via the API server.
```

Comment:
Since this is the client, it doesn't care (and know) where the configuration file is written to.

```suggestion
/// Configures network settings from the given input source (URI or stdin)
```